### PR TITLE
Distraction mechanic

### PR DESCRIPTION
**Pull request by jackh02**

### CODE REVIEW

File: Assets/Scripts/DistractionMechanic/DistractionMechanic.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+            if (playerCollider != null && throwableCollider != null)
+            {
+                Physics.IgnoreCollision(playerCollider, throwableCollider);
+            }
+            
+            Rigidbody throwableRigidbody = throwableObject.GetComponent<Rigidbody>();
+            throwableRigidbody.AddForce(transform.forward * throwForce, ForceMode.Impulse);
+
+            DistractionSource distractionSource = new DistractionSource(throwableObject.transform.position, 5f);
```
Comment (kit-calloway) [inline Assets/Scripts/DistractionMechanic/DistractionMechanic.cs]: This shouldn't be called immediately - instead, it should only be called when the `throwableRigidbody` registers a physics collision.

File: Assets/Scripts/DistractionMechanic/DistractionSource.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+{
+    public struct DistractionSource
+    {
+        public Vector3 Origin { get; }
+        private float _radius;
+
+        public DistractionSource(Vector3 origin, float radius)
+        {
+            this.Origin = origin;
+            this._radius = radius;
```
Comment (kit-calloway) [inline Assets/Scripts/DistractionMechanic/DistractionSource.cs]: Very minor, but the `this` keyword is redundant.

```suggestion
            Origin = origin;
            _radius = radius;
```

File: Assets/Scripts/DistractionMechanic/DistractionSource.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        {
+            Collider[] colliders = Physics.OverlapSphere(Origin, _radius);
+
+            foreach (Collider collider in colliders)
+            {
+                IDistractable distractable = collider.GetComponent<IDistractable>();
+                if (distractable != null)
+                {
+                    distractable.Distract(this);
+                }
```
Comment (kit-calloway) [inline Assets/Scripts/DistractionMechanic/DistractionSource.cs]: Braces can be removed

```suggestion
                if (distractable != null)
                    distractable.Distract(this);
```

File: Assets/Scripts/DistractionMechanic/Throwable.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace cmp2804
+{
+    public class Throwable : MonoBehaviour, IDistractable
+    {
+        public void Distract(DistractionSource source)
+        {
+            Debug.Log($"New distraction at {source.Origin}");
+        }
+    }
```
Comment (kit-calloway) [inline Assets/Scripts/DistractionMechanic/Throwable.cs]: The debug statement should be removed

```suggestion
    public class Throwable : MonoBehaviour
    {
    }
```

File: Assets/Scripts/DistractionMechanic/DistractionMechanic.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        {
+            GameObject throwableObject = Instantiate(throwable, transform.position, transform.rotation);
+
+            Collider playerCollider = GetComponent<Collider>();
+            Collider throwableCollider = throwableObject.GetComponent<Collider>();
+
+            if (playerCollider != null && throwableCollider != null)
+            {
+                Physics.IgnoreCollision(playerCollider, throwableCollider);
+            }
```
Comment (kit-calloway) [inline Assets/Scripts/DistractionMechanic/DistractionMechanic.cs]: Braces can be removed

```suggestion
            if (playerCollider != null && throwableCollider != null)
                Physics.IgnoreCollision(playerCollider, throwableCollider);
```

**Review by StevieGoodman — CHANGES_REQUESTED** [review]
Very good, just need to review my suggestions and make the distraction emit on collision with a rigid body.

File: Assets/Scripts/DistractionMechanic/DistractionMechanic.cs (hunk 1 above)
Comment (jackh02) [reply Assets/Scripts/DistractionMechanic/DistractionMechanic.cs]: Most recent commit should have resolved this issue but let me know if it needs to be changed

**Review by StevieGoodman — APPROVED** [review]
LGTM

**Pull request merged**